packages/bcer-api/app/src/location/locationDataPortal.controller.ts

```diff
@@ -394,61 +396,81 @@ export class LocationDataPortalController {
     await this.service.closeLocation([id], closedTime);
   }
 
-    /**
+  /**
    * 
    * Transfer a location to another business
   */
-    @ApiOperation({ summary: 'Transfer Location' })
-    @ApiResponse({ status: HttpStatus.OK })
-    @HttpCode(HttpStatus.OK)
-    @UseGuards(AuthDataGuard)
-    @Roles(ROLES.MOH_ADMIN, ROLES.HA_ADMIN)
-    @AllowAnyRole()
-    @Patch('/transfer/:locationId')
-    async transferLocation(
-      @Param('locationId') id: string,
-      @Query('businessId') businessId: string
-    ): Promise<void> {
-      if(!id)
-        throw new ForbiddenException("location ID is required");
+  @ApiOperation({ summary: 'Transfer Location' })
+  @ApiResponse({ status: HttpStatus.OK })
+  @HttpCode(HttpStatus.OK)
+  @UseGuards(AuthDataGuard)
+  @Roles(ROLES.MOH_ADMIN, ROLES.HA_ADMIN)
+  @AllowAnyRole()
+  @Patch('/transfer/:locationId')
+  async transferLocation(
+    @Param('locationId') id: string,
+    @Query('businessId') businessId: string
+  ): Promise<void> {
+    if(!id)
+      throw new ForbiddenException("location ID is required");
+
+    if(!businessId)
+      throw new ForbiddenException("New Business ID is required");
 
-      if(!businessId)
-        throw new ForbiddenException("New Business ID is required");
+    const location = await this.service.getLocation(id);    
+    const business = await this.businessService.getBusinessById(businessId); 
 
-      const location = await this.service.getLocation(id);    
-      const business = await this.businessService.getBusinessById(businessId); 
+    if (!location || !business) throw NotFoundException;
 
-      if (!location || !business) throw NotFoundException;
+    if (business.status !== BusinessStatus.Active)
+      throw new ForbiddenException('Cannot transfer location to a closed business')
+    
+    await this.service.transferLocation(id, businessId);
+  }
 
-      if (business.status !== BusinessStatus.Active)
-        throw new ForbiddenException('Cannot transfer location to a closed business')
-      
-      await this.service.transferLocation(id, businessId);
+  /**
+   * 
+   * Update fields of a location. (addressLine1 || postall || webpage || phone || email || underage || manufacturing || city || health_authority || longitude || latitude || geo_confidence)
+   */
+  @ApiOperation({ summary: 'Update fields of a Single Location' })
+  @ApiResponse({ status: HttpStatus.OK })
+  @HttpCode(HttpStatus.OK)
+  @Roles(ROLES.MOH_ADMIN, ROLES.HA_ADMIN)
+  @UseGuards(AuthDataGuard)
+  @AllowAnyRole()
+  @Patch('/update-fields/:locationId')
+  async editSingleLocation(
+    @Param('locationId') id: string,
+    @Body() payload: any

```

Comment:
What do you think about defining the type of this payload?

 type : {
   key: "addressLine1" || "postal" || "webpage" || phone || email || underage || manufacturing || city || health_authority || longitude || latitude || geo_confidence,
   value: string
}

or

type PossibleTypes =
  'addressLine1' | 'postal' | 'webpage' | 'phone' | 'email' | 'underage' |
  'manufacturing' | 'city' | 'health_authority' | 'longitude' | 'latitude' | 'geo_confidence';

interface LocationUpdatePayload {
  [key in PossibleTypes]?: string | boolean | number;
}